### PR TITLE
Bkammin/dsoda 163 google analytics

### PR DESCRIPTION
**Pull request by bkammin**

### CODE REVIEW

File: app/components/track-additional-info/template.hbs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {{#if (or @ensembleName @conductorName)}}
   <ToggleBox @verticalPosition={{@verticalPosition}} @horizontalPosition={{@horizontalPosition}} @open={{true}} @customIconPlacement={{true}} class="secondary-metadata-toggle-box" as |toggle|>
-    <toggle.trigger>
+    <toggle.trigger @class=" gtm__click-tracking-expand-track-info">
```
Comment (jkeen) [inline app/components/track-additional-info/template.hbs]: is the space in front of this intentional?

Comment (bkammin) [reply app/components/track-additional-info/template.hbs]: good catch

**Comment by bkammin** [comment]
@jkeen Good point -- I initially attempted this approach but the attributes didn't render, so I updated the component in `nypr-design-system` to handle the attributes as parameters here https://github.com/nypublicradio/nypr-design-system/pull/39/files

I dug into this a bit and realized that it's not working because share/link is set up as a "tag-less" component. This is accomplished by configuring it with an empty `tag` attribute here https://github.com/nypublicradio/nypr-design-system/blob/master/addon/components/nypr-m-share-tools/link.js#L10. It looks like if a component is tag-less, the component by default won't generate a HTML element like it would if the `tag` attribute was configured, and it just renders the template without its usual container.

In theory, I could refactor `share.link` to _not_ be tag-less, but there is currently a lot of customization going on in the template and it might be tricky to accomplish the refactor. Do you think this is worth tackling now?

**Comment by jkeen** [comment]
I dug in a bit too and wrestled with whether or not this is worth it, because changing things from tagless to tags would be a job, and it looks like _all_ those design system components are tagless. 

But then I discovered some good news! Looks like you can pass in attributes and assign them to another element using `...attributes`…meaning this would make it even easier to do what you wanted to do! 

All you'd have to change in the design system would be in that link template: 

```<MyComponent class="my-class" />```

And in that component, we can apply the class to the paragraph using ...attributes:
```
{{!--
  The paragraph gets the attributes, and not the h1
--}}
<h1>
  Hello, world!
</h1>
<p ...attributes>
  Lorem Ipsum...
</p>

template: 
```

So you could use this same technique for that design system change you did, where you could probably remove most of the stuff you added… and just add a `...attributes` to that element. I think that's the path to go

**Comment by jkeen** [comment]
This polyfill readme has good info on it (but we don't need the polyfill anymore since it's built in to ember now)

https://github.com/rwjblue/ember-angle-bracket-invocation-polyfill

**Comment by jkeen** [comment]
Actually we should add that `ember-angle-bracket-invocation-polyfill` polyfill to the design system, since we don't know the parent's app version that will be using the design systme addon.

File: app/components/track-additional-info/template.hbs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {{#if (or @ensembleName @conductorName)}}
   <ToggleBox @verticalPosition={{@verticalPosition}} @horizontalPosition={{@horizontalPosition}} @open={{true}} @customIconPlacement={{true}} class="secondary-metadata-toggle-box" as |toggle|>
-    <toggle.trigger>
+    <toggle.trigger @class="gtm__click-tracking-expand-track-info">
       {{yield (hash icon=toggle.icon)}}
```
Comment (jkeen) [inline app/components/track-additional-info/template.hbs]: missed one ^

**Pull request merged**